src/pylibsshext/logging.pyx

```diff
@@ -0,0 +1,101 @@
+#
+# This file is part of the pylibssh library
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see file LICENSE.rst in this
+# repository.
+
+import logging
+
+from pylibsshext.errors cimport LibsshSessionException
+
+
+ANSIBLE_PYLIBSSH_TRACE = 5
+
+LOG_MAP = {
+    logging.NOTSET: libssh.SSH_LOG_NONE,
+    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
+    logging.DEBUG: libssh.SSH_LOG_DEBUG,
+    logging.INFO: libssh.SSH_LOG_INFO,
+    logging.WARNING: libssh.SSH_LOG_WARN,
+    logging.ERROR: libssh.SSH_LOG_WARN,
+    logging.CRITICAL: libssh.SSH_LOG_WARN
+}
+
+LOG_MAP_REV = {
+    libssh.SSH_LOG_NONE: logging.NOTSET,
+    libssh.SSH_LOG_TRACE: ANSIBLE_PYLIBSSH_TRACE,
+    libssh.SSH_LOG_DEBUG: logging.DEBUG,
+    libssh.SSH_LOG_INFO: logging.INFO,
+    libssh.SSH_LOG_WARN: logging.WARNING,
+}
+
+logger = logging.getLogger("libssh")
+
+
+def add_trace_log_level():
```

Comment:
Could you explain why you're doing this many things in this function?

Reply:
Its some time since I wrote this so I do not remember where this was suggested exactly. I think some sources were suggesting that it might not be enough to do the `logging.addLevelName()`, but it might be enough for our use case. I will try if we can live without that.